bdsg/include/bdsg/snarl_distance_index.hpp

```diff
@@ -787,10 +785,185 @@ class SnarlDistanceIndex : public SnarlDecomposition, public TriviallySerializab
     }
 
 
+
+
+    
+//////////////////////////////////////////  The actual distance index
+    
+private:
+
+    /**
+     * This vector is the entire distance index. It is split up into "records" that are defined below
+     */
+    bdsg::yomo::UniqueMappedPointer<bdsg::MappedIntVector> snarl_tree_records;
+
+/*
+ * These are used to interpret snarl_tree_records, which is just a vector of ints
+ * Uses a SnarlTreeRecord as the main class for defining and interpreting the records
+ * SnarlTreeRecords are given a base pointer, which points to the start of the record
+ * Each record starts with a "tag", which defines which type of record it is. The 
+ * contents of the record are defined below: 
+*/
+    
+    /*Root record
+    
+     * - The (single) root vector has the format:
+     *   [root tag, # connected components (N), # nodes (M), min_node_id, max depth [pointer to node/snarl/chain record] x N], [pointer to node+ node offset] x M]
+     *   The root vector stores the root of every connected component, which can be a 
+     *   node, snarl, or chain
+     */
+    const static size_t ROOT_RECORD_SIZE = 5;
+    const static size_t COMPONENT_COUNT_OFFSET = 1;
```

Comment:
Should the offsets maybe be defined as the previous offset plus 1, and the size as the last offset plus 1, to make modification easier later, and to make sure none of the values can be wrong?

You might also be able to get away with making each of these "types" actually an enum, to let the compiler count for you.

Something like:
```
enum root_record_fields : int {
    ROOT_RECORD_TAG=0,
    COMPONENT_COUNT_OFFSET,
    NODE_COUNT_OFFSET,
    MIN_NODE_ID_OFFSET,
    MAX_TREE_DEPTH_OFFSET,
    ROOT_RECORD_SIZE
};
```

Also also, are you sure you don't want to consistently prefix all these names with the name of the type they belong to? If you see a line of code that says `record_start + NODE_COUNT_OFFSET`, it's not obvious that it really was supposed to be `record_start + SNARL_NODE_COUNT_OFFSET`.